### DISCUSSION TITLE
post/2020/2020-03-using-vscode-as-git-editor/

### DISCUSSION THREAD

**Discussion by giscus** [zyrikby/website_related#25] [Announcements]
# post/2020/2020-03-using-vscode-as-git-editor/

Last several years I use git as my version control system (VCS) both for personal and work projects. If you are working in a team, usage of a VCS brings you a lot of benefits like change tracking, history viewing, merge issues resolving, etc. However, I have found it very handy to use even for personal projects: with git I can try different solutions simultaneously and then select the better one. Moreover, I can easily remember what I have done for a project recently. This task becomes much easier if commit messages for repository changes are written clearly and in accordance with the patterns and rules recommended by the VCS. Developers have already developed the best practices how to write commit messages. I refer the interested reader, e.g., to the article written by Chris Beams. In this article, I explain how VSCode can help writing commit messages in accordance with the rules.

https://zhauniarovich.com/post/2020/2020-03-using-vscode-as-git-editor/

**Comment by atharvakadlag** [comment]
Why not use vim instead. 
nano wasnt great. but vim has been good for me for a long time now. 
any advantages of vscode over vim?
one disadvantage i can think of as you mentioned is the wait time.

**Reply by zyrikby** [reply]
VSCode is the default editor for me, therefore, I also use it for this task as well. I think in vim you can also configure rules (or add a plugin) that will check the format of the git message. So, answering your questions, there are no advantages besides that this is my default editor.
